compiler/arm-none-eabi-m4/compiler.yml

```diff
@@ -26,9 +26,9 @@ compiler.path.objsize: arm-none-eabi-size
 compiler.path.objcopy: arm-none-eabi-objcopy
 
 compiler.flags.base: -mcpu=cortex-m4 -mthumb-interwork -mthumb -Wall -Werror -fno-exceptions -ffunction-sections -fdata-sections
-compiler.flags.default: [compiler.flags.base, -O1 -ggdb]
-compiler.flags.optimized: [compiler.flags.base, -Os -ggdb]
-compiler.flags.debug: [compiler.flags.base, -Og -ggdb]
+compiler.flags.default: [compiler.flags.base, -O1, -ggdb]
+compiler.flags.optimized: [compiler.flags.base, -Os, -ggdb]
+compiler.flags.debug: [compiler.flags.base, -Og, -ggdb]
```

Comment:
Why must this be done? And why not do the same for m0/m3/m7/riscv?

Reply:
I mentioned this in the commit message, though admittedly I snuck it into this PR where it doesn't really belong.  It is better to separate the compiler flags because then newt can override the optimization level of a single package.  With the flags combined into a single item, newt doesn't recognize any of them as the "-O" flag.

Yes, we should make this same change to the other compiler definitions.